internal/testutils/backends.go

```diff
@@ -69,3 +69,29 @@ func BuildWinbindMock(t *testing.T, goPkgPath string) string {
 
 	return libPath
 }
+
+// BuildKrb5Mock takes the path to the location of the ad internal package and
+// builds the libkrb5 mock for use with package or integration tests.
+func BuildKrb5Mock(t *testing.T, goPkgPath string) string {
+	t.Helper()
+
+	filesToCompile := []string{filepath.Join(goPkgPath, "mock", "libkrb5_mock.c")}
+
+	// We only compile the krb5_init_context mock for package tests, as on
```

Comment:
Waow, well done!

However, I am not a big fan of the detection relying on "." itself. Do you think we can lookup (easily, don’t die over it) if we link against the real krb5 lib in the current process? (we can’t rely on the symbol itself: `krb5_init_context` from the krb5 lib will be probably be loaded after us, as we `LD_PRELOAD`). Maybe it’s too much, but this sounds a little bit too black magic to me :)

Reply:
Ok, fine then, I think as this is a testhelper, we can survive this (thans to the comment you have too) :)